sdk/iot/modelsrepository/src/constants.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft.
+// Licensed under the MIT license.
+
+const isBrowser = () => {
+  try {
+    return this === self;
+  } catch (e) {
+    return false;
+  }
+};
+const isNode = () => {
+  try {
+    return this === global;
+  } catch (e) {
+    return false;
+  }
+};
+
+const currentPlatform = isBrowser() ? "browser" : isNode() ? "node" : "undefined";
```

Comment:
out of curiosity, what do you do when `currentPlatform` is "undefined"? Also curious if it should be the actual value `undefined` instead of the string

Reply:
right now there is no real use for currentPlatform, which is probably why I let the bug on uncalled functions through. It is used to create a default user agent string, but the service is unmanaged so there's really no user agent string processing right now. This is just a future-proofing step.

Reply:
UserAgentPolicy can take care of platform detection:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-rest-pipeline/src/policies/userAgentPolicy.ts 

It already grabs the platform, bitness, etc

Reply:
It's part of the default pipeline returned by `createPipelineFromOptions`: https://github.com/Azure/azure-sdk-for-js/blob/3fcd69bc736592f5a4aee9f5588877cf3711d20b/sdk/core/core-rest-pipeline/src/pipeline.ts#L433

Or you can make an empty one and add it in if you don't want the other default behaviors